Parse/ParseClientConfiguration.m

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright (c) 2015-present, Parse, LLC.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import "ParseClientConfiguration.h"
+#import "ParseClientConfiguration_Private.h"
+
+#import "PFAssert.h"
+#import "PFApplication.h"
+#import "PFCommandRunningConstants.h"
+#import "PFFileManager.h"
+#import "PFHash.h"
+#import "PFObjectUtilities.h"
+
+@implementation ParseClientConfiguration
+
+///--------------------------------------
+#pragma mark - Init
+///--------------------------------------
+
++ (instancetype)emptyConfiguration {
+    return [[super alloc] initEmpty];
+}
+
+- (instancetype)initEmpty {
+    self = [super init];
+    if (!self) return nil;
+
+    _networkRetryAttempts = PFCommandRunningDefaultMaxAttemptsCount;
+
+    return self;
+}
+
+- (instancetype)initWithBlock:(void (^)(id<ParseMutableClientConfiguration>))configurationBlock {
+    self = [super init];
```

Comment:
Maybe invoke `-initEmpty` here and declare that as internal designated initializer (class extension in the .m file). Don't feel strongly about it, tho.